jax/_src/debugging.py

```diff
@@ -51,25 +56,37 @@ def debug_callback_abstract_eval(*flat_avals, callback: Callable[..., Any],
   del flat_avals, callback, in_tree
   return [], {effect}
 
-def debug_callback_batching_rule(*flat_args, callback: Callable[..., Any],
-    effect: DebugEffect, in_tree: tree_util.PyTreeDef):
-  del flat_args, callback, effect, in_tree
-  # TODO(sharadmv): implement batching rule
-  raise NotImplementedError('Batching not supported for `debug_callback`.')
+def debug_callback_batching_rule(args, dims, **params):
+  """Unrolls the debug callback across the mapped axis."""
+  axis_size = next(x.shape[i] for x, i in zip(args, dims)
+                   if i is not None)
+  # TODO(sharadmv): implement in terms of rolled loop unstead of
+  # unrolled.
+  def get_arg_at_dim(i, dim, arg):
+    if dim is batching.not_mapped:
+      # Broadcast unmapped argument
+      return arg
+    return lax.index_in_dim(arg, i, axis=dim, keepdims=False)
+  outs = []
+  for i in range(axis_size):
+    args_idx = map(functools.partial(get_arg_at_dim, i), dims, args)
+    outs.append(debug_callback_p.bind(*args_idx, **params))
+  outs = [jnp.stack(xs) for xs in zip(*outs)]
+  return outs, (0,) * len(outs)
 batching.primitive_batchers[debug_callback_p] = debug_callback_batching_rule
 
 def debug_callback_jvp_rule(*flat_args, callback: Callable[..., Any],
     effect: DebugEffect, in_tree: tree_util.PyTreeDef):
   del flat_args, callback, effect, in_tree
-  # TODO(sharadmv): implement jvp rule
-  raise NotImplementedError('JVP not supported for `debug_callback`.')
+  raise ValueError(
+      "JVP doesn't support debugging callbacks. "
+      "Instead, you can use them with `jax.custom_jvp` or `jax.custom_vjp`.")
 ad.primitive_jvps[debug_callback_p] = debug_callback_jvp_rule
 
 def debug_callback_transpose_rule(*flat_args, callback: Callable[..., Any],
     effect: DebugEffect, in_tree: tree_util.PyTreeDef):
   del flat_args, callback, effect, in_tree
-  # TODO(sharadmv): implement transpose rule
-  raise NotImplementedError('Transpose not supported for `debug_callback`.')
+  raise ValueError("Transpose doesn't support debugging callbacks.")
```

Comment:
I am wondering what are the cases when the user will see this message. For `vjp` and `grad` they will see the `jvp` message first. This error message makes sense if the user explicitly invokes transposition, e.g., `linear_transpose`. If there are other instances when this pops up, then maybe the error message should be more explanatory.